spatialos-sdk/src/worker/connection.rs

```diff
@@ -124,21 +125,34 @@ pub trait Connection {
 
     fn set_protocol_logging_enabled(&mut self, enabled: bool);
 
-    #[deprecated(note = "Use get_connection_status_code == ConnectionStatusCode::Success instead.")]
-    fn is_connected(&self) -> bool;
+    fn get_connection_status_code(&mut self) -> ConnectionStatusCode;
 
-    fn get_connection_status_code(&self) -> ConnectionStatusCode;
-    fn get_connection_status_detail(&self) -> String;
-    fn get_worker_id(&self) -> &str;
-    fn get_worker_attributes(&self) -> Vec<String>;
-    fn get_worker_flag(&self, name: &str) -> Option<String>;
+    fn get_connection_status_detail(&mut self) -> String;
```

Comment:
Consider replacing the `get_connection_status_code`/`detail` pair with a single `get_connection_status` that returns a `struct ConnectionStatus(ConnectionStatusCode, String)`. The C API will probably do this eventually anyway.